packages/flutter/lib/src/widgets/sensitive_content.dart

```diff
@@ -0,0 +1,225 @@
+
+  static final SensitiveContentSetting _instance = SensitiveContentSetting._();
+
+  /// Registers a [SensitiveContent] widget that will help determine the
+  /// [ContentSensitivity] level for the Flutter view with ID [viewId].
+  static void register(int viewId, ContentSensitivity desiredSensitivityLevel) {
+     _instance._register(viewId, desiredSensitivityLevel);
+  }
+
+  void _register(int viewId, ContentSensitivity desiredSensitivityLevel) {
+    if (!_contentSensitivityStates.containsKey(viewId)) {
+      _contentSensitivityStates[viewId] = ViewContentSensitivityState();
+    }
+
+    // Update SensitiveContent widget count for those with desiredSensitivityLevel.
+    final ViewContentSensitivityState contentSensitivityStateForView = _contentSensitivityStates[viewId]!;
+    contentSensitivityStateForView.addWidgetWithContentSensitivity(desiredSensitivityLevel);
+
+
+    // If only one SensitiveContent widget in the relevant view sets a
+    // ContentSensitivity level, then we can immediately set
+    // desiredSensitivityLevel for the view.
+    if (contentSensitivityStateForView.getTotalNumberOfWidgets() == 1) {
+      // Set content sensitivity level for view as desiredSensitivityLevel and update stored data.
+      _sensitiveContentService.setContentSensitivity(viewId, desiredSensitivityLevel);
+      contentSensitivityStateForView.currentContentSensitivitySetting = desiredSensitivityLevel;
+      return;
+    }
+
+    // Verify that desiredSensitivityLevel should be set in order for sensitive
+    // content in the view to remain secure.
+    if (!shouldSetContentSensitivity(currentSensitivityLevel: contentSensitivityStateForView.currentContentSensitivitySetting, desiredSensitivityLevel: desiredSensitivityLevel)) {
+      return;
+    }
+
+    // Set content sensitivity level for the view as desiredSensitivityLevel and update stored data.
+    _sensitiveContentService.setContentSensitivity(viewId, desiredSensitivityLevel);
+    contentSensitivityStateForView.currentContentSensitivitySetting = desiredSensitivityLevel;
+  }
+
+  /// Unregisters a [SensitiveContent] widget from the Flutter view with ID
+  /// [viewId].
+  static void unregister(int viewId, ContentSensitivity widgetSensitivityLevel) {
+     _instance._unregister(viewId, widgetSensitivityLevel);
+  }
+
+  void _unregister(int viewId, ContentSensitivity widgetSensitivityLevel) {
+    // Update SensitiveContent widget count for those with
+    // desiredSensitivityLevel.
+    final ViewContentSensitivityState contentSensitivityStateForView = _contentSensitivityStates[viewId]!;
+    contentSensitivityStateForView.removeWidgetWithContentSensitivity(widgetSensitivityLevel);
+
+    if (contentSensitivityStateForView.getTotalNumberOfWidgets() == 0) {
+      // There is no longer sensitive content in the view. Reset to the default
+      // mode.
+      _sensitiveContentService.setContentSensitivity(viewId, ContentSensitivity.autoSensitive);
```

Comment:
Nit: Maybe the default ContentSensitivity should be saved in a constant somewhere and referenced here. I know it's used in the initialization of currentContentSensitivitySetting too.